src/pages/SignIn.jsx

```diff
@@ -1,20 +1,51 @@
-import React from 'react';
+import React, {useContext, useEffect, useState} from 'react';
 import { Link } from 'react-router-dom';
+import {AuthContext} from '../context/AuthContext';
+import axios from "axios";
 
 function SignIn() {
-  return (
-    <>
-      <h1>Inloggen</h1>
-      <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ab alias cum debitis dolor dolore fuga id molestias qui quo unde?</p>
+    const {setUser, getUser, Toggle, getSecret, username, state} = useContext(AuthContext);
+    const [usrnm, setUsrnm] = useState("");
 
-      <form>
-        <p>*invoervelden*</p>
-        <button>Inloggen</button>
-      </form>
+    useEffect(() => {
+        setUsrnm(localStorage.getItem('username'))
+    }, [localStorage.getItem('username')]);
 
-      <p>Heb je nog geen account? <Link to="/signup">Registreer</Link> je dan eerst.</p>
-    </>
-  );
+    async function handleSubmit(e) {
+        e.preventDefault();
+
+        try {
+            const response = await axios.post('http://localhost:3000/login', {
+                email: document.getElementById("email").value,
+                password: document.getElementById("password").value,
+            });
+            localStorage.setItem('token', response.data.accessToken);
+        } catch (e) {
+            console.error(e);
+        }
+
+        setUser();
+        getSecret();
+        Toggle();
+    }
+
+    return (
+        <main>
+            {!state &&<> <h1>Inloggen</h1>
+            <form>
```

Comment:
Het is gebruikelijk om bij een formulier in de form tag de handleSubmit functie aan te roepen, bij de button in plaats van type="button", type="submit" mee te geven.
```suggestion
            <form onsubmit={handleSubmit}>
```